airflow/providers/google/marketing_platform/operators/campaign_manager.py

```diff
@@ -20,16 +20,23 @@
 """
 import tempfile
 import uuid
+import warnings
 from typing import Any, Dict, List, Optional
 
 from googleapiclient import http
 
 from airflow.exceptions import AirflowException
 from airflow.models import BaseOperator
-from airflow.providers.google.cloud.hooks.gcs import GCSHook
 from airflow.providers.google.marketing_platform.hooks.campaign_manager import GoogleCampaignManagerHook
 from airflow.utils.decorators import apply_defaults
 
+try:
+    from airflow.providers.google.cloud.hooks.gcs import GCSHook
+except ImportError:
+    warnings.warn("The package google-cloud must be installed to use the operator.",
```

Comment:
In this case I think we should bundle gcp and gmp as all "report" operators will require GCS or BQ hooks.

Reply:
I think this is still OK as it is IMHO. I want to make it a good case to test multiple packages and I would not make an exception here. What we could do is to make intra-dependencies and automatically install google.cloud when gmp is used. We should make some consistent rules and stick to it - if we want to group some packages, we should group them. If we want to split - we should keep them separated.

Reply:
In case of gmp we heavily use GooglsCloudBaseHook for hooks

Reply:
I agree - but either we decide to move it all to one folder/provider or we leave it as three dependent packages. I do not want to make an exception to the structure of provider packages. It makes it much easier to have one approach where provider package (e.g. google.cloud) is under "providers/google/cloud/" directory and "operators/hooks/sensors" are right directly under it. 

This way we do not have to keep any mapping between folders and provider packages - it is straightforward and can be easily derived from where the operators/hooks/sensors folders are (in fact I am already doing it when I check for import-cross-dependencies).

I am ok with both -  separate packages and dependencies between them and moving everything back to "google" provider. Alternatively we could have google/common but I think it's the worst idea of the three. I have slight preference for option 1) because with limited risk and effort it will let us explore consequences of having multiple packages with dependencies but 2) is also fine for me.

Whatever we do - I think we should keep consistency between folder structure and "provider packages" structure (provider name is formed from the directories between "providers" and "operators" basically).

What do you think @mik-laj  @nuclearpinguin @michalslowikowski00  @ANiteckiP @kaxil @ashb ?

Option 1):
1 single "google" provider package:
```
providers/google/
                             operators
                             hooks
                             sensors 
```
Option 2)
3 separate providers: "google.cloud", "google.marketing", "google.suite". Both "gogle.marketing" and "google.suite" is pretty much useless without "google.cloud" (but it will be printed as warning when there is no google.cloud already installed.. 

```
providers/google/cloud
                                     operators
                                    ...
providers/google/marketing
                                         operators
                                         ...
providers/google/suite
                                    operators
```

Reply:
1️⃣ is my prefered choice

Reply:
BTW, If you look at https://travis-ci.org/apache/airflow/jobs/653861799?utm_medium=notification&utm_source=github_status - you will se the kind of static check errors that are printed when one provider package imports directly another provider without try/except clause. There are just 63 cases left and I am happy to fix them all once we agree that this is good idea and right approach of handling cross-provider-package dependencies.

Reply:
+1 for 1, in this way users don't have to care under which Google umbrella service is

Reply:
@mik-laj @ANiteckiP @michalslowikowski00 @kaxil ?

Reply:
+1 for 1

Which would mean `cncf.kubernets` becomes `cncf`, `amazon.aws` becomes `amazon`, ...
Should the name we remove from the packages be added to the name of the hook/operator/sensor ? `aws_athena`, `aws_batch`, `aws_cloud_formation`, ... ?

Reply:
In my opinion the first level in the `providers` package is the provider. All the packages within are just names used in the context of the provider.

I would suggest to only consider the first package as a provider that can be installed via a seperate provider package.

Reply:
So actually.. now I want to change my "vote" to +1 for 2. And make sure that we define it like this "The first level is the provider that can be installed via a provider package which can contain sub-domains".

WDYT?

Reply:
> Which would mean `cncf.kubernets` becomes `cncf`, `amazon.aws` becomes `amazon`, ...

Not necessarily - I think we can still keep some sub-packages for those packages that have little cross dependencies. For example there are nearly no cross dependencies in "providers/apache" ... 

The above discussion is strictly about "google" top-level provider and whether to split it or combine.I think it could be a decision made individually for each "top-level" provider - whether to split further or keep everything at one level. I think that's perfectly fine to have both options available.

Reply:
BTW - this is still following what we already agreed to in AIP-21 - the rule about whether to split or not was left up to the top-level provider.

Reply:
> So actually.. now I want to change my "vote" to +1 for 2. And make sure that we define it like this "The first level is the provider that can be installed via a provider package which can contain sub-domains".

My thought was that it would be much better to have "operators"/"hooks"/"sensors" always staring as the "first" level of the provider. It think it makes much more sense to have all "operators" of the provider in one directory rather than split.

Reply:
OK. I found an easy way - i can use the hashmap from the setup_backport_opertors to define what are the packages. Without adding any extra mapping. This way we can still keep the "google" related packages in a single "backport package" but with internal "cloud/marketing/suite" structure. Updated PR shortly and I will make all the try/except corrections and rebase to make it ready for review.